object-filter/src/main/java/org/infinispan/objectfilter/impl/syntax/KnnPredicate.java

```diff
@@ -40,12 +40,24 @@ public ValueExpr getChild() {
 
    @Override
    public String toString() {
-      return leftChild.toString() + "<-> " + vector + "~" + knn;
+      return leftChild.toString() + " <-> " +
```

Comment:
This might as well create a StringBuilder and call appendQueryString as well.

Reply:
Also why different with the vector param? Is that a different issue?

Reply:
I noticed that the `toString` is implemented independently in all the other predicates. So I would like to make this guy consistent with all the others. I guess we don't really need an efficient `toString`.
About the issue I think that the scope of [ISPN-1592](https://issues.redhat.com/browse/ISPN-15925) is about to fix the sub-query string generated from the knn predicate, while the change we made for [ISPN-15932](https://issues.redhat.com/browse/ISPN-15932) improve the query text generation of all predicates, so it has a different goal. For that reason I opened a different issue.

Reply:
I would still recommend doing. While it may not seem like much, if the leftChild string is large it will still be a large String copy operation.

```java
StringBuilder sb = new StringBuilder();
leftChild.appendQueryString(sb);
sb.append(" <-> ").append(vector != null ? vector : "[" + vectorParam + "]").append('~').append(knn.toString());
```

Reply:
For me it is the same. I pushed the change to make the to string using the builder as well.

Reply:
For the JVM it isn't ;) But this looks even better than I thought 👍